File: src/utils/file.utils.js
const fs = require("fs");
const path = require("path");

/**
 * Create a directory recursively. No-op if it already exists.
 */
function createFolder(dirPath) {
  fs.mkdirSync(dirPath, { recursive: true });
}

/**
 * Create a file with content. Creates parent directories if needed.
 */
function createFile(filePath, content) {
  const dir = path.dirname(filePath);
  createFolder(dir);
  fs.writeFileSync(filePath, content, "utf-8");
}

module.exports = { createFolder, createFile };

File: aura_mcp/utils/file_utils.py
"""Safe filesystem helpers for project scaffolding."""

from __future__ import annotations

from pathlib import Path


def create_folder(dir_path: str | Path) -> None:
    """Create a directory tree. No-op if it already exists."""
    Path(dir_path).mkdir(parents=True, exist_ok=True)


def create_file(file_path: str | Path, content: str) -> None:
    """Write *content* to *file_path*, creating parent directories as needed."""
    path = Path(file_path)
    path.parent.mkdir(parents=True, exist_ok=True)
    path.write_text(content, encoding="utf-8")
